main.nf

```diff
@@ -94,14 +94,91 @@ if (params.help) {
 }
 
 // Validate inputs
-if(!params.raw_input){
-   params.mzmls = params.mzmls ?: { log.error "No spectra data privided. Make sure you have used the '--mzmls' option."; exit 1 }()
+//MS Input
+if (params.sample_sheet)  {
+   sample_sheet = file(params.sample_sheet)
+
+   Channel.from( sample_sheet )
+                .splitCsv(header: true, sep:'\t')
+                .map { col -> tuple("${col.ID}", "${col.Sample}", "${col.Condition}", file("${col.ReplicateFileName}", checkifExists: true))}
+                .flatMap{it -> [tuple(it[0],it[1].toString(),it[2],it[3])]}
+                .into { ch_samples_from_sheet; ch_samples_for_fasta}
+
+}
+
+if(params.mzml_input){
```

Comment:
Having two flags `mzml_input` and `raw_input` seems unnecessary and bares the risk if the user specifying both, which is not caught as far as I can tell. I would suggest to make one behaviour the default, e.g. raw files, and add a flag `--from_mzml` which switches the behaviour.

Reply:
ok, ill fix this in hands with the --spectra input comment from Lukas and Timo